### DISCUSSION TITLE
Power off safely

### DISCUSSION THREAD

**Discussion by rolb15** [probonopd/MiniDexed#584] [Q&A]
How do you perform a safe power off with minidexed?
I have pulled the power cord and ruined the sd-card, there must be a better way.
Connect a button to gpio? A command?

**Comment by Banana71** [comment]
Hi @rolb15,

I don't have a solution for you, I can only tell you what I consider. The SD card is read during the boot phase (approx.
8 seconds with 20 voice banks). While loading and saving performances, the SD card is accessed for a short time (<0,5s) . During these phases, switching off the power supplies should be avoided. Knowing this, I have been switching the miniDexed on and off via the power supply several times a day for more than 1.5 years without any problems. It could be that I was just lucky, but I haven't heard of a similar problem before.

Maybe that will help: https://circle-rpi.readthedocs.io/_/downloads/en/44.4/pdf/

**Reply by rolb15** [reply]
Thanks!
Maybe it was just one bad sd-card or I somehow managed to turn off during the boot phase.

**Comment by plamen-dzhorev** [comment]
Hi @rolb15,

I also never had issues using the power supply. My device is battery powered from a buck converter directly on the 5V pin on the Pi. You can create backups each time you insert the card into your PC for updates - this is more reliable than implementing power off button.

**Comment by diyelectromusic** [comment]
The issue with safe removal under Windows or properly powering off/unmounting under Linux is related to the complex software layers that sit between a card and the applications.  MiniDexed, running on circle, has none of that.

With all the layers there is a significant possibility of partial updates to the cards still sitting around in memory having not made it out to the card itself.  This is why a proper shutdown/etc is required - it forces everything into a consistent state prior to losing power.  MiniDexed doesn't need any of that.  In fact the times when it writes anything at all to the card are pretty minimal - even configuration changes are not preserved back into the minidexed.ini file.

The only time I could imagine having an issue is if you attempting to save/update/delete a performance and then immediately killed power before it could finish.  But even then there is more chance of just ending up with a corrupt performance than an inconsistent disk, although if you got really "lucky" and caught an update of the file tables then maybe...  There may be others, but that is the only one that springs to mind for me.

Flash is a little more complex, especially serial flash such an SD card, but again if you're not writing, then there is nothing to go wrong :)

In short, I'm much more included to imagine a faulty card than a software corruption problem due to not shutting down.

Kevin

**Comment by probonopd** [comment]
How do you "power off safely" any synthesizer on the market that is running a CPU? You just switch it off, right? MiniDexed should not require to specifically "power off safely", or we are doing it wrong. Of course powering off the device during a write operation should be avoided, but otherwise it should be safe to power off at any time.

Should there be a reproducible issue, please open an issue ticket which we will treat as a bug. Thanks!
